Kzl 969 cpp websocket ctor (#197)
kuzzleio/documentation-V2: src/sdk-reference/cpp/1/websocket/constructor/index.md

---
layout: sdk.html.hbs
title: Constructor

> description: Create a new Websocket object connected to the backend

Reviewer: ```suggestion
 description: Create a new Websocket object connected to Kuzzle server```

order: 0
---

# Constructor

WebSocket is a class implementing the virtual class [Protocol]({{site_base_path}}sdk-reference/cpp/1/protocol).

This is the main entry point to communicate with Websocket protocol.

> Give an instance of WebSocket to Kuzzle to connect to it.

Reviewer 2: ```suggestion
 Pass an instance of this class to the [constructor of the Kuzzle SDK]({{site_base_path}}sdk-reference/cpp/1/kuzzle/constructor/) to use this protocol to connect to the Kuzzle server.
```

## Signature

```cpp

> WebSocket(const std::string& host, protocol, kuzzleio::options *options = nullptr)

Reviewer: We should prefer multi signature instead of default parameters

> WebSocket(const std::string& host, protocol, kuzzleio::options *options = nullptr)

Reviewer: ```suggestion
 WebSocket(const std::string& host, kuzzleio::options *options = nullptr)
```

```

## Arguments

| Argument  | Type        | Description                     |
| --------- | ----------- | ------------------------------- |

> | `host`    | <pre>const std::string&</pre> | Kuzzle host you want to connect to |

Reviewer: ```suggestion
 | `host`    | <pre>const std::string&</pre> | Kuzzle server hostname or IP |
```

> | `options` | <pre>kuzzleio::options\*</pre>   | WebSocket connection configuration |

Reviewer: ```suggestion
 | `options` | <pre>kuzzleio::options\*</pre>   | WebSocket protocol configuration |
```

> ### **options**

Reviewer: ```suggestion
 ### options
```

> | Option               | Type               | Description                                                        | Default  | 

Reviewer: ```suggestion
 | Option               | Type<br/>(`default`)               | Description                                                        |
```

> | -------------------- | ------------------ | ------------------------------------------------------------------ | -------- |

Reviewer: ```suggestion
 | -------------------- | ------------------ | ------------------------------------------------------------------ |
```

> | `port`     | int      | The connection port                 | `7512`

Reviewer: ```suggestion
 | `port`     | <pre>int</pre><br/>(`7512`)      | Kuzzle server port              |
```

> | `ssl`    | bool      | Use ssl to connect to the kuzzle instance              | `false`

Reviewer: ```suggestion
 | `ssl`    | <pre>bool</pre><br/>(`false`)      | Use SSL to connect to Kuzzle server              |
```
